compiler/pash_runtime.py

```diff
@@ -116,7 +117,7 @@ def compile_optimize_output_script(ir_filename, compiled_script_file, args, comp
     ## which should be translated to a parallel script.
     if(isinstance(optimized_ast_or_ir, IR)):
         if args.distributed_exec:
-            ir_filename = os.path.join(config.PASH_TMP_PREFIX, 'distributed_exec_ir.pkl')
+            ir_filename = os.path.join(config.PASH_TMP_PREFIX, f'{str(uuid4())[:8]}_distributed_exec_ir.pkl')
```

Comment:
I think you should use this `https://github.com/binpash/pash/blob/distribution-bug-fixes/compiler/util.py#L42` instead of manually creating random file names.

Reply:
The reason I'm not using it is that when debugging (using -d) I want to keep the file but ptempfile always gets deleted when the pash finishes.

Reply:
I thought that when `-d` is used PASH_TMP_PREFIX is not deleted (https://github.com/binpash/pash/blob/main/pa.sh#L79), which would by the way affect your solution too.

Furthermore, ptempfile uses tempfile.mkstemp which is not deleted automatically and requires manual cleanup as far as I know.

Reply:
Double check and you are right, changed it to use ptempfile.